static/js/Header.jsx

```diff
@@ -393,6 +401,346 @@ SearchBar.propTypes = {
+      if (document.getElementById('keyboardInputMaster')) {
+        return; // if keyboard is open, ignore
+      }
+      if (Sefaria.interfaceLang === 'english' && !props.hideHebrewKeyboard) {
+        const keyboardInitiator = document.querySelector(".keyboardInputInitiator");
+        if (keyboardInitiator) {
+          keyboardInitiator.style.display = show ? "inline" : "none";
+        }
+      }
+    };
+    const focusSearch = () => {
+      setSearchFocused(true);
+      showVirtualKeyboardIcon(true);
+    };
+
+    const blurSearch = (e) => {
+      onBlur(e)
+      const parent = document.getElementById('searchBox');
+      if (!parent.contains(e.relatedTarget) && !document.getElementById('keyboardInputMaster')) {
+        setSearchFocused(false);
+        showVirtualKeyboardIcon(false);
+      }
+    };
+
+    const inputClasses = classNames({
+      search: 1,
+      serif: 1,
+      keyboardInput: Sefaria.interfaceLang === "english",
+      hebrewSearch: Sefaria.interfaceLang === "hebrew"
+    });
+
+    const searchBoxClasses = classNames({ searchBox: 1, searchFocused });
+
+
+    return (
+      <div id="searchBox"
+           className={searchBoxClasses}>
+        <SearchButton onClick={handleSearchButtonClick} />
+        <input
+          className={inputClasses}
+          id="searchInput"
+          placeholder={Sefaria._("Search")}
+          onKeyDown={handleSearchKeyDown}
+          onFocus={focusSearch}
+          onBlur={blurSearch}
+          maxLength={75}
+          title={Sefaria._("Search for Texts or Keywords Here")}
+          {...otherDownShiftProps}
+        />
+      </div>
+    );
+  };
+const SuggestionsDispatcher = ({ suggestions, getItemProps, highlightedIndex}) => {
+  function groupByType(seggestedItems) {
+    const groupedItems = {};
+
+    // Group items by their "type", "Topic" and "PersonTopic" considered same type
+    seggestedItems.forEach(item => {
+        let itemType = item.type in ["Topic", "PersonTopic"] ? "Topic" : item.type;
+        if (!groupedItems[itemType]) {
+            groupedItems[itemType] = [];
+        }
+        groupedItems[itemType].push(item);
+    });
+
+    //Convert into a datastructure like this: [{"type": name,
+    //                                         "items" : [item1, item2]}]
+    const result = Object.keys(groupedItems).map(type => ({
+        type,
+        items: groupedItems[type]
+    }));
+
+    return result;
+}
+
+    let groupedSuggestions = groupByType(suggestions);
+    let universalIndex = 0;
+
+    return (
+        <>
+            {groupedSuggestions.map((object, index) => {
+                const UniversalInitialIndexForGroup = universalIndex;
+                universalIndex += object.items.length;
+                return (
+                    <SuggestionsGroup
+                        getItemProps={getItemProps}
+                        highlightedIndex={highlightedIndex}
+                        key={index}
```

Comment:
The `key` should be the type of the group, not the index. Using the index is considered an anti-pattern in React.